Core/src/Surfaces/LineSurface.cpp

```diff
@@ -151,45 +141,51 @@ Acts::SurfaceIntersection Acts::LineSurface::intersect(
     const GeometryContext& gctx, const Vector3& position,
     const Vector3& direction, const BoundaryCheck& bcheck,
     ActsScalar tolerance) const {
-  // following nominclature found in header file and doxygen documentation
-  // line one is the straight track
+  // The nomenclature is following the header file and doxygen documentation
+
   const Vector3& ma = position;
   const Vector3& ea = direction;
-  // line two is the line surface
-  const auto& tMatrix = transform(gctx).matrix();
-  Vector3 mb = tMatrix.block<3, 1>(0, 3).transpose();
-  Vector3 eb = tMatrix.block<3, 1>(0, 2).transpose();
-  // now go ahead and solve for the closest approach
-  Vector3 mab(mb - ma);
+
+  // Origin of the line surface
+  Vector3 mb = transform(gctx).translation();
+  // Line surface axis
+  Vector3 eb = transform(gctx).rotation() * Vector3::UnitZ();
```

Comment:
The extraction + transpose is probably cheaper than the full matrix multiplication, right?

Reply:
I don't know this, but it's possible, yes.